src/tutorials/0006/02.md

```diff
@@ -0,0 +1,35 @@
+By using a cryptographic algorithm to generate the hash, we guarantee that we can identify a file with a single hash.
+
+Occassionally, a hashing algorithm may stop being compliant with the previously stated rules, as has already happened with `sha1`. With time, other algorithms may prove to be insufficient for content addressing in IPFS. For this reason, and in order to support multiple cryptographic algorithms, **we need to be able to know which algorithm was used to generate the hash** of specific content.
+
+![What is the hashing algorithm used in a hash?](tutorial-assets/T0006L02-what-algo.jpg)
+
+So how can we do this?
+To support multiple hashing algorithms, we use **Multihash**.
+
+## Multihash format
+
+A [**multihash**](https://multiformats.io/multihash/) is a self-describing hash which itself contains metadata that describes both its length and what cryptographic algorithm generated it. IPFS is future-proof because it uses multihash to
+**support multiple hashing algorithms** rather than relying on a specific one.
+
+Multihashes follow the `TLV` pattern (`type-length-value`). Essentially, the "original hash" is prefixed with a `type` and the `length` of the hash.
+
+![Multihash format](tutorial-assets/T0006L02-multihash.jpg)
+
+ - `type`: identifier of the **cryptographic algorithm** used to generate the hash (e.g. the identifier of `sha2-256` would be `18` - `0x12` in hexadecimal. see the [multicodec table](https://github.com/multiformats/multicodec/blob/master/table.csv) for all the identifiers)
+ - `length`: the actual **length** of the hash (using `sha2-256` it would be `256`)
+ - `value`: the actual **hash value**
+
+To allow representation of a CID as a compact string (instead of plain binary), IPFS uses `base58btc` encoding to get the following:
+
+`QmY7Yh4UquoXHLPFo2XbhXkhBvFoPwmQUSa92pxnxjQuPU`
+
+So with this full specification, the first version of IPFS CIDs was born! This first version is called Version 0 (`CIDv0`), and the initial `Qm...` characters are easily spotted on its CIDs.
+
+However, with time, doubts about whether this format would be sufficient started to arise:
+
+ - How do we know the encoding of the data?
```

Comment:
I think this could do with a little more explanation. I see it is explained further below but there's no mention of encoded data prior to this sentence.

Reply:
@terichadbourne how do you think we could go about this?